library/data/registry_entry_attachments/src/main/java/org/quiltmc/qsl/registry/attachment/impl/RegistryEntryAttachmentSync.java

```diff
@@ -134,25 +154,33 @@ private static void fillEncodedValuesCache() {
 									.formatted(attachment.id(), valueEntry.getKey()));
 						}
 
-						myTable.put(entryId.getNamespace(), entryId.getPath(), attachment.codec()
+						encoded.computeIfAbsent(entryId.getNamespace(), id -> new HashSet<>()).add(
+								new AttachmentEntry(entryId.getPath(), false, attachment.codec()
 								.encodeStart(NbtOps.INSTANCE, valueEntry.getValue())
 								.getOrThrow(false, msg -> {
 									throw new IllegalStateException("Failed to encode value for attachment %s of registry entry %s: %s"
 											.formatted(attachment.id(), entryId, msg));
-								}));
+								})));
+					}
+				}
+
+				Map<TagKey<Object>, Object> entryTagValues = dataHolder.valueTagTable.rowMap().get(attachmentEntry.getValue());
+				if (entryTagValues != null) {
+					for (var valueEntry : entryTagValues.entrySet()) {
+						encoded.computeIfAbsent(valueEntry.getKey().id().getNamespace(), id -> new HashSet<>()).add(
+								new AttachmentEntry(valueEntry.getKey().id().getPath(), true, attachment.codec()
+										.encodeStart(NbtOps.INSTANCE, valueEntry.getValue())
+										.getOrThrow(false, msg -> {
+											throw new IllegalStateException("Failed to encode value for attachment tag %s of registry %s: %s"
+													.formatted(attachment.id(), valueEntry.getKey().id(), msg));
+										})));
 					}
 				}
 
-				Set<Pair<String, NbtCompound>> valueMaps = new HashSet<>();
-				for (var tableEntry : myTable.rowMap().entrySet()) {
-					/*var valueMap = new NbtCompound();
-					for (var valueEntry : tableEntry.getValue().entrySet()) {
-						valueMap.put(valueEntry.getKey(), valueEntry.getValue());
-					}*/
+				Set<Pair<String, Set<AttachmentEntry>>> valueMaps = new HashSet<>();
+				for (var namespaceEntry : encoded.entrySet()) {
 					// this is probably a horrible idea lmao
```

Comment:
what

Reply:
Pretty sure this is just a weird diff? Seems like most of the code relating to this was rewritten anyway.

Reply:
> // this is probably a horrible idea lmao

Reply:
oh yeah, I wrote that because the syncing code I wrote is probably horrible.